docs/kb/rules/using-semgrep-rule-schema-in-vscode.md

```diff
@@ -0,0 +1,78 @@
+
+You may already be familiar with writing rules in the [Semgrep Playground](/docs/playground). However, if your IDE of choice is VS Code and you'd like to write Semgrep rules there, using the Semgrep rule schema will provide a richer editing environment, allowing VS Code to understand the shape of your rule's YAML file, including its value sets, defaults, and descriptions ([reference](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas)).
+
+:::tip
+Writing rules locally in your IDE is also helpful for iteratively testing your rule against an entire local repository, as opposed to just a snippet of test code.
+:::
+
+When the schema is set up, auto-completion operates in your VS Code IDE just as it does in the Semgrep Playground when writing rules:
+
+![Example Semgrep YAML rule file with auto-complete](/img/kb/vscode-schema-autocomplete-example.png)
+
+## Adding the Semgrep rule schema in VS Code
+
+Adding the Semgrep rule schema in VS Code requires two steps:
+
+1. Install the YAML Language Support extension by Red Hat
+2. Associate the Semgrep rule schema
+
+### Install the YAML Language Support extension by Red Hat
+
+You can install the extension either directly in VS Code or by going to the Visual Studio Marketplace and installing it from there. In VS Code, typically just searching `yaml` will yield the correction extension as the top result. The extension is named "YAML" and is authored by "Red Hat". However, please also verify you are installing the correct extension by ensuring it is the same as the one [linked here](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).
+
+### Associate the Semgrep rule schema
+
+Once the extension is installed, associate the Semgrep rule schema with Semgrep YAML rule definitions you are working on in VS Code using one of two recommended methods:
+
+1. Directly in the YAML file
+2. Using `yaml.schemas` in your VS Code `settings.json` file
+
+We recommend taking a look at the [extension overview section on associating schemas](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas) as a precursor to gain a preliminary understanding before proceeding.
+
+#### Associating a schema directly in the YAML file
+
+To associate the schema easily and directly with a Semgrep YAML rule file, include the following line at the top of the file:
+
+    # yaml-language-server: $schema=https://json.schemastore.org/semgrep.json
+
+The drawback to this method is that it must be done independently for each YAML rule file.
+
+#### Associating a schema to a glob pattern via `yaml.schemas`
+
+Again, we highly recommend first reading the [extension overview](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-a-schema-to-a-glob-pattern-via-yaml.schemas) as a supplement to this article to better understand how YAML schemas are handled by the extension.
```

Comment:
```suggestion
Before proceeding, Semgrep recommends that you read the [extension overview](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-a-schema-to-a-glob-pattern-via-yaml.schemas) as a supplement to this article to better understand how the extension handles YAML schemas.
```

Reply:
I still feel weird about us saying this. If people need to read the extension docs to do this, maybe it's not a method we should actually recommend. I know if I saw this I'd ignore it, and I even like reading docs.

Reply:
I'm not sure I agree. When I was doing this, I found it helpful to read the extension overview page, specifically the schema section, before actually taking any action. Similarly I think users would find it beneficial to read over that, then follow our specific instructions with that background knowledge in mind.